src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/Cells/DiaryDayEntryTableViewCell.swift

```diff
@@ -4,19 +4,196 @@
 
 import UIKit
 
-class DiaryDayEntryTableViewCell: UITableViewCell {
+class DiaryDayEntryTableViewCell: UITableViewCell, UITextFieldDelegate {
+
+	// MARK: - Protocol UITextFieldDelegate
+
+	func textFieldDidEndEditing(_ textField: UITextField) {
+		cellModel.updateCircumstances(textField.text ?? "")
+	}
+
+	func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+		endEditing(true)
+	}
 
 	// MARK: - Internal
 
-	func configure(cellModel: DiaryDayEntryCellModel) {
+	func configure(
+		cellModel: DiaryDayEntryCellModel,
+		onInfoButtonTap: @escaping () -> Void
+	) {
+		self.cellModel = cellModel
+		self.onInfoButtonTap = onInfoButtonTap
+
 		checkboxImageView.image = cellModel.image
 		label.text = cellModel.text
+		label.font = cellModel.font
+
+		addParameterViews()
+
+		parametersContainerStackView.isHidden = cellModel.parametersHidden
+
+		durationSegmentedControl.selectedSegmentIndex = cellModel.selectedDurationSegmentIndex
+		maskSituationSegmentedControl.selectedSegmentIndex = cellModel.selectedMaskSituationSegmentIndex
+		settingSegmentedControl.selectedSegmentIndex = cellModel.selectedSettingSegmentIndex
+		visitDurationPicker.date = Date.dateWithMinutes(cellModel.locationVisitDuration) ?? Date()
+		notesTextField.text = cellModel.circumstances
+
 		accessibilityTraits = cellModel.accessibilityTraits
+
+		let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(headerTapped))
+		headerStackView.addGestureRecognizer(tapGestureRecognizer)
+		headerStackView.isUserInteractionEnabled = true
```

Comment:
In reuse cases this may lead to duplicate added views and gesture recognizers

Reply:
`addParameterViews()` actually removes them first. I renamed it to `setupParameterViews()` to make it clearer now.

Good point on the gesture recognizer. I'm adding it in `awakeFromNib()` now so it's not added multiple times 😊